ipatests/pytest_ipa/integration/tasks.py

```diff
@@ -139,9 +139,14 @@ def wrapped(*args, **kwargs):
 
 def prepare_reverse_zone(host, ip):
     zone = get_reverse_zone_default(ip)
-    result = host.run_command(["ipa",
-                      "dnszone-add",
-                      zone], raiseonerr=False)
+    result = host.run_command(
+        [
+            "ipa",
+            "dnszone-add",
+            "--skip-overlap-check",  # to force dnszone-add
+            zone
+        ],
+        raiseonerr=False)
```

Comment:
Hi @ndehadrai 
I don't think this diff would fix the issue. If you look for instance at the logs for PR#3103:
- during the setup for `test_replica_promotion.py::TestUnprivilegedUserPermissions`, `ipa-server-install` is called with `--auto-reverse` and it creates the reverse zone 121.168.192.in-addr.arpa. (see [log](http://freeipa-org-pr-ci.s3-website.eu-central-1.amazonaws.com/jobs/9768c652-6c67-11e9-b4ec-fa163ef01e1e/runner.log.gz)) because the server has IP@ 192.168.121.74
- during the test `test_replica_promotion.py::TestUnprivilegedUserPermissions::()::test_replica_promotion_by_unprivileged_user`, the test calls `ipa dnszone-add 121.168.192.in-addr.arpa.` (see [log](http://freeipa-org-pr-ci.s3-website.eu-central-1.amazonaws.com/jobs/9768c652-6c67-11e9-b4ec-fa163ef01e1e/report.html)) because the client has IP@ 192.168.121.116
So the test is trying to create a reverse zone that already exists. The option `--skip-overlap-check` would force the creation even if the zone was handled *by another DNS server*, not if it is already handled by IPA.
A correct fix would rather be to check first if the zone is already defined in IPA, and try to add only if it's not the case, or ignore the error `DNS zone with name "xxx" already exists`.